proto/config/netinst.proto

```diff
@@ -137,11 +143,14 @@ message NetworkInstanceConfig {
   // routes can be propagated at the same time, there are no restrictions for using both.
   //
   // Note that the default route (with the bridge IP as the gateway) is automatically
-  // propagated to connected applications unless explicitly disabled by setting
-  // NetworkInstanceConfig.ip.gateway to an all-zeroes IP or when the uplink is app-shared
-  // (not management) and does not have a default route of its own. In the latter case,
-  // it is possible to enforce default route propagation by configuring a static default
-  // route for the network instance.
+  // propagated to connected applications with these exceptions:
+  //  a) default route propagation is explicitly disabled by setting
+  //     NetworkInstanceConfig.ip.gateway to an all-zeroes IP
+  //  b) network instance is air-gapped (without uplink)
+  //  c) the uplink is app-shared (not management) and does not have a default route
+  //     of its own
+  // In the b) and c) cases, it is possible to enforce default route propagation
```

Comment:
What happens if I add a static default route in case a)?  Do you plan to explicitly check and reject/ignore that in the code?

Reply:
Yes, the plan was for the zero gateway to take priority.